app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/DisplayDatasetActivity.kt

```diff
@@ -68,22 +83,25 @@ class DisplayDatasetActivity : AppCompatActivity() {
     }
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        return when (item?.itemId) {
+        val categoriesArray = ArrayList<Category>(categories)
+        return when (item.itemId) {
             R.id.display_dataset_menu_modify_categories -> {
                 val intent = Intent(this@DisplayDatasetActivity, DataCreationActivity::class.java)
-                //TODO: Give to the DataCreationActivity the list of the categories of the dataset
-                //intent.putExtra("dataset_categories", dummyPresenter.getCategories())
+                intent.putParcelableArrayListExtra("dataset_categories", categoriesArray)
+                intent.putExtra("dataset_id", datasetId)
                 startActivity(intent)
                 true
             }
+            //Clicked on Add new Picture button
             else -> {
```

Comment:
That's what I had before but If I do that I can't (Or I don't see how) to Test this "else" (because the else is needed anyway) which lowers the  coverage for no good reason.